packages/toolpad-app/pages/deploy/[appId]/[[...path]].tsx

```diff
@@ -14,6 +17,20 @@ export const getServerSideProps: GetServerSideProps<ToolpadAppProps> = async (co
     };
   }
 
+  const app = await getApp(appId);
+  if (!app) {
+    return {
+      notFound: true,
+    };
+  }
+
+  if (!app.public) {
+    if (!checkBasicAuth(context.req, context.res)) {
```

Comment:
would it be possible to throw an error and make it trigger the proper response?

Reply:
maybe nevermind this comment I made - i guess throwing errors is not the right away of returning error responses in Next.js serverless functions.

Reply:
I'm planning to split this into two functions. one that checks the request headers and one that sets the response headers. that should make it explicit enough.